File: 515. Find Largest Value in Each Tree Row.py
class Solution:
    def largestValues(self, root: Optional[TreeNode]) -> List[int]:
        store = {}
        def DFS(root, depth):
            if not root: return
            if depth in store:
                store[depth] = max(store[depth], root.val)
            else: store[depth] = root.val
            DFS(root.left, depth + 1)
            DFS(root.right, depth + 1)

        DFS(root, 0)
        ans = [0] * len(store)
        for depth in store:
            ans[depth] = store[depth]
        return ans

File: 515. Find Largest Value in Each Tree Row.cpp
class Solution {
public:
    unordered_map<int, int> store, visited;
    void DFS(TreeNode* root, int depth)
    {
        if(not root) return;
        if(not visited[depth]) store[depth] = root->val;
        else store[depth] = max(store[depth], root->val);
        visited[depth] = true;
        DFS(root->left, depth + 1);
        DFS(root->right, depth + 1);
    }
    vector<int> largestValues(TreeNode* root) 
    {
        DFS(root, 0);
        vector<int> ans(store.size());
        for(auto [depth, value] : store)
            ans[depth] = value;
        return ans;
    }
};